### PR TITLE
feat: observed data likelihood strategy

### LINKED ISSUE #60: DOCS: add documentation about data observed Likelihood strategy
Describe the essence of the strategy, provide references to sources, and prove that the strategy can be implemented component by component

### LINKED ISSUE #107: FEAT: Add generalized observed data Likelihood strategy
Add a generic implementation of the strategy.

### PR DESCRIPTION
**Pull request by iraedeus**
This pull request introduces a new optimization strategy for mixture models: maximizing the observed data likelihood directly. The main changes add support for this strategy throughout the codebase, including the implementation, integration into the strategy selection system, and comprehensive unit tests to ensure correctness and robustness.

**Observed Data Likelihood Strategy Implementation:**

- Added the `observed_data_likelihood_strategy` function, which computes optimized parameters for mixture components by maximizing the observed data log-likelihood. This includes careful handling of numerical stability and background contributions from other components.

**Integration and API Updates:**

- Registered the new strategy in the strategy selection system, updating the `__init__.py` exports and the mapping in `MaximizationStep` to allow users to select it via the `MaximizationStrategy` enum. [[1]](diffhunk://#diff-77e77de6bc4ef59d523ddba7633a74b48d698fb0f947e818f41ac15d7ba8d9a5R16-R32) [[2]](diffhunk://#diff-61235073011965b7b0ed918299e3260fba3f6e495378678f7ba32ae1a6a64aa9L26-R26) [[3]](diffhunk://#diff-61235073011965b7b0ed918299e3260fba3f6e495378678f7ba32ae1a6a64aa9R69)
- Added an `OBSERVED_DATA_LIKELIHOOD` member to the `MaximizationStrategy` enum, with documentation describing its purpose.

**Testing and Documentation:**

- Added a comprehensive unit test suite for the new strategy, covering mathematical correctness, parameter handling, optimizer interaction, and edge cases such as single-component mixtures and fixed parameters.
- Provided a mathematical justification for the observed data likelihood optimization in a new LaTeX documentation file, detailing the derivation and optimization conditions.

### CODE REVIEW

File: rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        L(θ_q) = Σ ln( C_i + w_q * f_q(x_i | θ_q) )
+
+    Where C_i is the fixed contribution of all other components (background)
+    calculated based on the current state.
+
+    Parameters
+    ----------
+    component : ContinuousDistribution[DType]
+        The distribution component type/instance used for dispatch and parameter metadata.
+    state : PipelineState
```
Comment (xImoZA) [inline rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py]:
```suggestion
    state : PipelineState[DType]
```

File: rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    Returns
+    -------
+    tuple[int, dict[str, DType]]
+        Component ID and a dictionary of the optimized parameters.
+    """
+
+    X = state.X
+    n_samples = X.shape[0]
+    dtype = component.dtype
+    tol = dtype(NUMERICAL_TOLERANCE)
```
Comment (xImoZA) [inline rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py]: I've been thinking about this, and it seems better to use `np.finfo(dtype).tiny` because `np.float16(1e-9)` incorrectly evaluates to zero.

File: rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    temp_mixture = copy(state.curr_mixture)
+    target_comp = temp_mixture.components[component_id]
+    background_term = np.zeros(n_samples, dtype=dtype)
+
+    for i, comp in enumerate(state.curr_mixture.components):
+        if i != component_id:
+            background_term += temp_mixture.weights[i] * comp.pdf(X)
+
+    background_term = np.maximum(background_term, tol)
```
Comment (xImoZA) [inline rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py]: Is clipping the background_term necessary here? Numerical stability `log(0)` is already handled by clipping the mixture_pdf

File: rework_tests/unit/estimators/iterative/_strategies/test_observed_data_likelihood.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+import numpy as np
+import pytest
+from rework_pysatl_mpest.core import MixtureModel, Parameter
+from rework_pysatl_mpest.distributions import ContinuousDistribution
+from rework_pysatl_mpest.estimators.iterative import MaximizationStrategy, OptimizationBlock, PipelineState
+from rework_pysatl_mpest.estimators.iterative._strategies import observed_data_likelihood_strategy
+from rework_pysatl_mpest.optimizers import Optimizer
+
+DTYPES_TO_TEST = [np.float32, np.float64]
```
Comment (xImoZA) [inline rework_tests/unit/estimators/iterative/_strategies/test_observed_data_likelihood.py]:
```suggestion
DTYPES_TO_TEST = [np.float16, np.float32, np.float64]
```
All tests passed successfully.

File: rework_tests/unit/estimators/iterative/_strategies/test_observed_data_likelihood.py
Comment (xImoZA) [inline rework_tests/unit/estimators/iterative/_strategies/test_observed_data_likelihood.py:1]: Missing test case for handles_numerical_overflow.

File: rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+__author__ = "Danil Totmyanin"
+__copyright__ = "Copyright (c) 2025 PySATL project"
+__license__ = "SPDX-License-Identifier: MIT"
+
+
+from copy import copy
+
+import numpy as np
+
+from rework_pysatl_mpest.estimators.iterative._strategies.utils import handle_numerical_overflow
```
Comment (xImoZA) [inline rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py]:
```suggestion
from .utils import handle_numerical_overflow
```

File: rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+from ....distributions import ContinuousDistribution
+from ....optimizers import Optimizer
+from ....typings import DType
+from ..pipeline_state import PipelineState
+from ..steps import OptimizationBlock
+
+NUMERICAL_TOLERANCE = 1e-9
```
Comment (xImoZA) [inline rework_pysatl_mpest/estimators/iterative/_strategies/observed_data_likelihood.py]: Unused constant

**Review by xImoZA — DISMISSED** [review]
LGTM

File: rework_tests/unit/estimators/iterative/_strategies/test_observed_data_likelihood.py [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+    Verifies that the original component object in the mixture is not modified
+    in place before the optimization result is applied (strategy works on copies).
+    """
+
+    target_comp, state, block, optimizer, _ = parametrized_setup
+    original_param1 = target_comp.param1
+
+    observed_data_likelihood_strategy(target_comp, state, block, optimizer)
+
+    assert target_comp.param1 == original_param1
```
Comment (VITYANA) [inline rework_tests/unit/estimators/iterative/_strategies/test_observed_data_likelihood.py:124]: is it correct? you mocked Optimizer so it's instantly returns optimizer.minimize.return_value = [dtype(5.0), dtype(5.0)] -> target never calls -> target_comp.set_params_from_vector never calls -> the mutation never happens during the test execution

**Review by xImoZA — DISMISSED** [review]
